### ISSUE TITLE
Number formatter not coping with BigDecimal floats

### ISSUE THREAD

**Issue by opoudjis** [plurimath/plurimath#262]
````
Plurimath::NumberFormatter.new(:de).
localized_number("14236.39239", format: {:digit_count=>19}) == 
"14.236,39239000000000"

Plurimath::NumberFormatter.new(:de).
localized_number("327428.7432878432992", format: {:digit_count=>19}) == 
"327.428"
````

This is happening because lib/plurimath/formatter/numbers/fraction.rb digit_count_format() is using Float. We must not use Float. We must use BigDecimal.

**Comment by opoudjis** [comment]
Submitted PR. Please review and release; please add following tests:

````
num1 = "327428.000878432992"
puts Plurimath::NumberFormatter.new(:de).localized_number(num1, format: {:digit_count=>18})
puts Plurimath::NumberFormatter.new(:de).localized_number(num1, format: {:digit_count=>19})
puts Plurimath::NumberFormatter.new(:de).localized_number(num1, format: {:digit_count=>20})
puts Plurimath::NumberFormatter.new(:de).localized_number(num1, format: {:digit_count=>18, notation: :scientific})
puts Plurimath::NumberFormatter.new(:de).localized_number(num1, format: {:digit_count=>19, notation: :scientific})
puts Plurimath::NumberFormatter.new(:de).localized_number(num1, format: {:digit_count=>20, notation: :scientific})

````

**Comment by suleman-uzair** [comment]
Fixed in #263
